__tests__/app.test.js

```diff
@@ -306,7 +306,70 @@ describe("GET /api/reviews/:review_id/comments", () => {
       .then((res) => {
         const comment = res.body.comment;
         expect(Array.isArray(comment)).toBeTruthy();
-        expect(comment.length).toBe(0)
+        expect(comment.length).toBe(0);
+      });
+  });
+});
+describe.only("10-POST /api/reviews/:review_id/comments", () => {
+  test("should post the comment with the relevent review id", () => {
+    return request(app)
+      .post("/api/reviews/4/comments")
+      .send({
+        username: "mallionaire",
+        body: "oooooooooh",
+      })
+      .expect(201)
+      .then((res) => {
+        const comment = res.body.comment;
+        expect(comment).toEqual(
+          expect.objectContaining({
+            review_id: 4,
+            comment_id: expect.any(Number),
+            body: "oooooooooh",
+            author: "mallionaire",
+            votes: expect.any(Number),
+            created_at: expect.any(String),
+          })
+        );
+      });
+  });
+  test("should return 404 and a msg if review_id not found", () => {
+    return request(app)
+      .post("/api/reviews/9999/comments")
+      .send({
+        username: "mallionaire",
+        body: "oooooooooh",
+      })
+      .expect(404)
+      .then((res) => {
+        const msg = res.text;
+        expect(msg).toBe("Sorry can't find that!");
+      });
+  });
+  test("should return 400 and a msg if review_id of wrong type", () => {
+    return request(app)
+      .post("/api/reviews/haha/comments")
+      .send({
+        username: "mallionaire",
+        body: "oooooooooh",
+      })
+      .expect(400)
+      .then((res) => {
+        const msg = res.text;
+        expect(msg).toBe("invalid type please check your input");
+      });
+  });
+  test("should return 404 and a msg if username not found", () => {
+    return request(app)
+      .post("/api/reviews/4/comments")
+      .send({
+        username: "lala",
+        body: "oooooooooh",
+      })
+      .expect(404)
+      .then((res) => {
+        const msg = res.text;
+        expect(msg).toBe("Sorry can't find that!");
```

Comment:
Great tests so far, should add one more 400 for if there are missing fields in the comment username or body